### PR TITLE
Auth: Remove out-of-zone-additional-processing setting

### LINKED ISSUE #6461: documentation clarification or feature removal  of out-of-zone-additional-processing
- Program: Authoritative
 - Issue type: Not 100% sure, may be a documentation bug report or a 'feature removal request'

Sorry for not following the issue template exactly here, I think in this case it is not 100% applicable

### Environment
PowerDNS authoritative 4.1 (and probably earlier) documentation

### Short description
While reading through the documentation we stumbled upon `out-of-zone-additonal-processing` ( https://doc.powerdns.com/md/authoritative/settings/#out-of-zone-additional-processing ) which currently  reads

> out-of-zone-additional-processing
> Boolean
> Default: yes
> Do out of zone additional processing. This means that if a malicious user adds a '.com' zone to your server, it is not used for other domains and will not contaminate answers. Do not enable this setting if you run a public DNS service with untrusted users.
> The docs had previously indicated that the default was "no", but the default has been "yes" since 2005.

This is confusing, as the default value is `yes` but the text says the operator should **not enable** this setting.   I assumed this was simply a typo and created PR #6459 which changes `Do not enable` to `Do not disable`. 

However, for example the Openstack designate documentation recommends setting this to `no` : 
 * https://docs.openstack.org/designate/3.0.1/production-guidelines.html#powerdns-mitigation

So it is currently unclear from the documentation, which setting actually prevents the attack (`yes` or `no` )

The comment by @Habbie in the PR and discussion in IRC concluded that this feature should maybe be removed altogether.

### PR DESCRIPTION
**Pull request by pieterlexis**
### Short description
This PR removes the `out-of-zone-additional-processing` option and thereby "defaults" to not ever adding additional data that is out of zone to the additional section.

Closes #6461

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [x] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/packethandler.cc [hunk 1]
```diff
@@ -457,18 +457,17 @@ int PacketHandler::doAdditionalProcessingAndDropAA(DNSPacket *p, DNSPacket *r, c
         lookup = getRR<NSRecordContent>(i->dr)->getNS();
       else
         continue;
+
       B.lookup(QType(d_doIPv6AdditionalProcessing ? QType::ANY : QType::A), lookup, p);
```
Comment (mind04) [inline pdns/packethandler.cc]: use  B.lookup(QType(d_doIPv6AdditionalProcessing ? QType::ANY : QType::A), lookup, p, sd.domain_id) here and you can skip the id and partof check below.

Comment (pieterlexis) [reply pdns/packethandler.cc]: Did that, thank you!

**Comment by pieterlexis** [comment]
It looks like the LDAP backend does not use ID's and hence adds additional data, so we might need the `isPartOf` check regardless.

```
[2018-11-09 19:33:07] +++ ./tests/no-out-of-zone-data/real_result	2018-11-09 19:33:06.420297581 +0000
[2018-11-09 19:33:07] @@ -1,3 +1,4 @@
[2018-11-09 19:33:07]  0	_imap._tcp.example.com.	IN	SRV	120	0 1 143 blah.test.com.
[2018-11-09 19:33:07] +2	blah.test.com.	IN	A	3600	192.168.6.1
[2018-11-09 19:33:07]  Rcode: 0 (No Error), RD: 0, QR: 1, TC: 0, AA: 1, opcode: 0
[2018-11-09 19:33:07]  Reply to question for qname='_imap._tcp.example.com.', qtype=SRV
```

**Comment by Habbie** [comment]
This pull request **introduces 1 alert** when merging 87c261bcddd25cea34885c4b49bb88641ed24c9e into 26b34c5fb7a1139a18eb5335dc2ea5b49a60c26f - [view on LGTM.com](https://lgtm.com/projects/g/PowerDNS/pdns/rev/pr-0704e1ee0101907075f40ff6cfbb468b0436a371)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**